### DISCUSSION TITLE
Modifying an implicit surface

### DISCUSSION THREAD

**Discussion by eflehder1** [leap71/PicoGK#48] [Coding for (Computational) Engineers]
@josefinelissner @LinKayser could you advice me on how I could go about creating further modifications to the surface of an implicitly generated geometry such as that shown below? Is there a way to create for instance a pattern of bumps throughout the surface? 
![image](https://github.com/user-attachments/assets/53c96456-702d-4a91-8b15-fec77ee5218b)

**Comment by LinKayser** [comment]
Hi @eflehder1 — if you are already generating the surface implicitly, then the most elegant way is to modify the implicit formula to add the bumps. You could use PerlinNoise or a similar function to come up with modifications to the signed distance you are calculating in the implicit function.

Other ways would be converting to a mesh and then modifying the vertex coordinates. But your challenge lends itself nicely to a modification of the SDF.

Simply adding a Perlin noise value to the signed distance you calculate seems promising to me (but I haven't tried it).

Best,
Lin

**Reply by eflehder1** [reply]
Thank you @LinKayser I will try that! I was hoping to find a more controlled way of creating a surface pattern, something similar to the controlled way in which one creates modulations with the shapekernel for base shapes. However, this alternative would be interesting so I will play around with the SDF as you suggest!

**Reply by LinKayser** [reply]
If you are looking for something specific, happy to give you guidance on that. You can easily convert to a mesh and then apply transformations to the vertices. The other thing you can do is paint on top of the surface with lattices, for example (use spheres to drill indentations, etc.).

Lin

**Reply by eflehder1** [reply]
thank you for the quick response @LinKayser will look into these strategies! At this stage my intention was just to get an understanding for how this could be done. In some situations it might not be viable to increase surface area by reducing the pore size of the lattice, it could then be interesting to explore using controlled patterns at the micro-scale. 

Many thanks,
Eric
